crypto/merklearray/proof.go

```diff
@@ -96,3 +98,33 @@ func (p *SingleLeafProof) GetConcatenatedProof() []byte {
 	}
 	return proofconcat
 }
+
+// ProofDataToSingleLeafProof receives serialized proof data and uses it to construct a proof object.
+func ProofDataToSingleLeafProof(hashTypeData string, treeDepth uint64, proofBytes []byte) (SingleLeafProof, error) {
+	hashType, err := crypto.UnmarshalHashType(hashTypeData)
+	if err != nil {
+		return SingleLeafProof{}, err
+	}
+
+	var proof SingleLeafProof
+
+	proof.HashFactory = crypto.HashFactory{HashType: hashType}
+	proof.TreeDepth = uint8(treeDepth)
+
+	digestSize := proof.HashFactory.NewHash().Size()
+	if len(proofBytes)%digestSize != 0 {
+		return SingleLeafProof{}, fmt.Errorf("proof bytes length is %d, which is not a multiple of "+
```

Comment:
please use the error wrapping technique: take a look here:
https://github.com/algorand/go-algorand/blob/792f2b652a70b943b99185aef1c58065e8a10968/crypto/merklearray/merkle.go#L371
you can define the new error at `merkle.go`